cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1651,6 +1688,58 @@ void writer::impl::write(table_view const& table, std::vector<partition_info> co
       task.wait();
     }
   }
+
+  if (stats_granularity_ == statistics_freq::STATISTICS_COLUMN) {
+    // need pages on host to create offset_indexes
+    thrust::host_vector<gpu::EncPage> h_pages = cudf::detail::make_host_vector_async(pages, stream);
+    stream.synchronize();
+
+    // add column and offset indexes to metadata
+    for (auto b = 0, r = 0; b < static_cast<size_type>(batch_list.size()); b++) {
+      auto const rnext   = r + batch_list[b];
+      auto curr_page_idx = chunks[r][0].first_page;
+      for (; r < rnext; r++) {
+        int p           = rg_to_part[r];
+        int global_r    = global_rowgroup_base[p] + r - first_rg_in_part[p];
+        auto& row_group = md->file(p).row_groups[global_r];
+        for (auto i = 0; i < num_columns; i++) {
+          gpu::EncColumnChunk& ck = chunks[r][i];
```

Comment:
I think this can be const.

```suggestion
          gpu::EncColumnChunk const& ck = chunks[r][i];
```

Reply:
yeah, I still have to get my head around using const more religiously.  it's natural in scala where it's the default (val vs var).  There are a whole lot of places to clean that up.